code/datums/forced_movement.dm

```diff
@@ -0,0 +1,98 @@
+//Just new and forget
+/datum/forced_movement
+	var/atom/movable/victim
+	var/atom/target
+	var/last_processed
+	var/steps_per_tick
+	var/allow_tabling
+	var/spin
+															//as fast as ssfastprocess
+/datum/forced_movement/New(atom/movable/avictim, atom/atarget, asteps_per_tick = 0.5, aallow_tabling = FALSE, aspin = FALSE)

```

Comment:
I must say, i'd really rather prefer something like src.victim = avictim. adding a to the front just makes it harder to read at a glance.

up to you.